src/util/locking/RedisLocker.ts

```diff
@@ -0,0 +1,225 @@
+ * (one lua script is considered one operation.)
+ */
+export class RedisLocker implements ReadWriteLocker, ResourceLocker, Finalizable {
+  protected readonly logger = getLoggerFor(this);
+
+  private readonly redis: Redis;
+  private readonly redisRw: RedisReadWriteLock;
+  private readonly redisLock: RedisResourceLock;
+  private readonly attemptSettings: Required<AttemptSettings>;
+
+  public constructor(redisClient = '127.0.0.1:6379', attemptSettings: AttemptSettings = {}) {
+    this.redis = this.createRedisClient(redisClient);
+    this.attemptSettings = { ...attemptDefaults, ...attemptSettings };
+
+    // Register lua scripts
+    this.redis.defineCommand('acquireReadLock', {
+      numberOfKeys: 1,
+      lua: SCRIPTS.acquireReadLock,
+    });
+    this.redis.defineCommand('acquireWriteLock', {
+      numberOfKeys: 1,
+      lua: SCRIPTS.acquireWriteLock,
+    });
+    this.redis.defineCommand('releaseReadLock', {
+      numberOfKeys: 1,
+      lua: SCRIPTS.releaseReadLock,
+    });
+    this.redis.defineCommand('releaseWriteLock', {
+      numberOfKeys: 1,
+      lua: SCRIPTS.releaseWriteLock,
+    });
+    this.redis.defineCommand('acquireLock', {
+      numberOfKeys: 1,
+      lua: SCRIPTS.acquireLock,
+    });
+    this.redis.defineCommand('releaseLock', {
+      numberOfKeys: 1,
+      lua: SCRIPTS.releaseLock,
+    });
+
+    // @ts-expect-error Use custom interface type including the ReadWrite Lock operations just added via lua.
+    this.redisRw = this.redis;
+    // @ts-expect-error Use custom interface type including the Resource Lock operations just added via lua.
+    this.redisLock = this.redis;
+  }
+
+  /**
+   * Generate and return a RedisClient based on the provided string
+   * @param redisClientString - A string that contains either a host address and a
+   *                            port number like '127.0.0.1:6379' or just a port number like '6379'.
+   */
+  private createRedisClient(redisClientString: string): Redis {
+    if (redisClientString.length > 0) {
+      // Check if port number or ip with port number
+      // Definitely not perfect, but configuring this is only for experienced users
+      const match = /^(?:([^:]+):)?(\d{4,5})$/u.exec(redisClientString);
+      if (!match || !match[2]) {
+        // At least a port number should be provided
+        throw new Error(`Invalid data provided to create a Redis client: ${redisClientString}\n
+            Please provide a port number like '6379' or a host address and a port number like '127.0.0.1:6379'`);
+      }
+      const port = Number(match[2]);
+      const host = match[1];
+      return new Redis(port, host);
+    }
+    throw new Error(`Empty redisClientString provided!\n
+            Please provide a port number like '6379' or a host address and a port number like '127.0.0.1:6379'`);
+  }
+
+  /**
+   * Try a redis function according to the set {@link AttemptSettings}
```

Comment:
```suggestion
   * Try a Redis function according to the set {@link AttemptSettings}
```
I might have missed some but make sure to be consistent in Redis capitalization.